docs/design/simple-rolling-update.md

```diff
@@ -0,0 +1,84 @@
+## Simple rolling update
+This is a lightweight design document for simple rolling update in ```kubectl```
+
+### Lightweight rollout
+Assume that we have a current replication controller named ```foo``` and it is running image ```image:v1```
+
+```kubectl rolling-update rc foo [foo-v2] --image=myimage:v2```
+
+If the user doesn't specify a name for the 'next' controller, then the 'next' controller is renamed to
+the name of the original controller.
+
+Obviously there is a race here, where if you kill the client between delete foo, and creating the new version of 'foo' you might be surprised about what is there, but I think that's ok.
+See Recovery below
+
+If the user does specify a name for the 'next' controller, then the 'next' controller is retained with its existing name,
+and the old 'foo' controller is deleted.
+
+#### Recovery
+If a rollout fails or is terminated in the middle, it is important that the user be able to resume the roll out.
+To facilitate recovery in the case of a crash of the updating process itself, we add the following annotations to each replicaController:
+   * ```desired-replicas``` The desired number of replicas for this controller (either N or zero)
+   * ```update-partner``` A pointer to the resource that is the other half of this update (syntax ```<namespace>:<id>```)
+
+Recovery is achieved by issuing the same command again:
+
+```
+kubectl rolling-update rc foo [foo-v2] --image=myimage:v2
+```
+
+Whenever the rolling update command executes, the kubectl client looks for replication controllers called ```foo``` and ```foo-next```, if the exist, an attempt is
+made to roll ```foo``` to ```foo-next```.  If ```foo-next``` does not exist, then it is created, and the rollout is a new rollout.  If ```foo``` doesn't exist, then
+it is assumed that the rollout is nearly completed, and ```foo-next``` is renamed to ```foo```.  Details of the execution flow are given below.
+
+
+### Aborting a rollout
+Abort is assumed to want to reverse a rollout in progress.
+
+```kubectl rolling-update rc foo [foo-v2] abort```
```

Comment:
It's confusing that "abort" is a magic keyword, and will even be more confusing once we support rollover. It would need to be a flag. However, --rollback would make more sense to me.